src/Language/OpenSCAD/Writer.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Language.OpenSCAD.Writer where
+import Language.OpenSCAD
+import qualified Data.ByteString.Char8 as BS
+import Data.Maybe (fromMaybe)
+
+write :: [TopLevel] -> [BS.ByteString]
+write = map writeTopLevel
+
+writeTopLevel :: TopLevel -> BS.ByteString
+writeTopLevel x = case x of
+    TopLevelScope obj    -> writeObject True obj
+    UseDirective str     -> BS.pack ("use <" ++ str ++ ">")
+    IncludeDirective str -> BS.pack ("include <" ++ str ++ ">")
+
+writeObject :: Bool -> Object -> BS.ByteString
+writeObject isTopLevel obj = case obj of
+     Module (Ident ident) args maybeObj ->
+        BS.pack (ident ++ "()")
+        `BS.append` fromMaybe BS.empty (fmap (BS.append (BS.pack " ") . writeObject False) maybeObj)
+        `BS.append` if isTopLevel then BS.pack ";" else BS.empty
```

Comment:
This is definitely a suboptimal use of `ByteString` as repeated `append` has quadratic complexity. At very least we should use `Data.ByteString.Builder`. Even better would be `Data.Text.Lazy.Builder` (since OpenSCAD source can contain Unicode characters IIRC).

However, the best option by far would be to use a proper pretty-printer library (e.g. `prettyprinter`).